### PR TITLE
sync GH actions with IUC

### PR DESCRIPTION
**Pull request by bernt-matthias**
seems that the PR workflow does not work anymore, because of `set-env`. Guess its about time to sync the workflows again.

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -203,7 +238,7 @@ jobs:
     - name: lintr
       run: |
         library(lintr)
-        linters <- with_defaults(line_length_linter = NULL, cyclocomp_linter = NULL)
+        linters <- with_defaults(line_length_linter = NULL, object_usage_linter = NULL)
```
Comment (nsoranzo) [inline .github/workflows/pr.yaml:241]: Should we keep `cyclocomp_linter = NULL` (added in commit 4e17a8df806c4f37c83874ebf46b97c924586a5a )?

Comment (bernt-matthias) [reply .github/workflows/pr.yaml:241]: Probably yes and we should also add it to the IUC workflow, or?

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -396,27 +413,25 @@ jobs:
...
       with:
         path: ~/.cache/pip
-        key: pip_cache_py_${{ matrix.python-version }}_gxy_${{ env.GALAXY_HEAD_SHA }}
+        key: pip_cache_py_${{ matrix.python-version }}_gxy_${{ needs.setup.outputs.galaxy_head_sha }}
     - name: Install Planemo
       run: pip install planemo
     - name: Deploy on testtoolshed
       env:
-        SHED_KEY: ${{ secrets.tts_api_key }}
+        SHED_KEY: ${{ secrets.TTS_API_KEY }}
```
Comment (bernt-matthias) [inline .github/workflows/pr.yaml:426]: @nsoranzo do you know if the change to uppercase requires an update of the settings?

Comment (nsoranzo) [reply .github/workflows/pr.yaml:426]: Yes, it does need someone with admin privileges on this repo to remove and re-add the TTS and TS API key secrets.

File: .github/workflows/pr.yaml (hunk 1 above)
Comment (nsoranzo) [reply .github/workflows/pr.yaml:241]: I suppose it would make life easier on tools-iuc too.

Comment (bernt-matthias) [reply .github/workflows/pr.yaml]: done. and https://github.com/galaxyproject/tools-iuc/pull/3398

File: .github/workflows/pr.yaml (hunk 2 above)
Comment (bernt-matthias) [reply .github/workflows/pr.yaml:424]: OK. Given the activity on this repo we can ask for this change after this is merged.

Comment (martenson) [reply .github/workflows/pr.yaml:424]: I doubt this is needed, we have uppercase version in the secrets since June

Comment (bernt-matthias) [reply .github/workflows/pr.yaml:424]: I remember that they are shown upper case even if entered lower case. But we can just try with one of my long open PRs :)

Comment (martenson) [reply .github/workflows/pr.yaml:424]: @bernt-matthias lol, in that case I have no idea what we have. I am ready to change it if it fails though.

**Comment by bernt-matthias** [comment]
Ready to go?

**Comment by bernt-matthias** [comment]
Seems that deployment works https://github.com/galaxyproject/tools-devteam/actions/runs/451752521

**Pull request merged**